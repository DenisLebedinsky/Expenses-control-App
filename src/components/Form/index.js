import React, { useState } from 'react'

export default ({ addItem }) => {
  const [expenses, setExpenses] = useState(0)
  const [incomes, setIncomes] = useState(0)
  const [name, setName] = useState('')

  const id = Math.floor(Math.random() * (1000 - 1) + 1)

  const _addItem = () => {
    if (typeof addItem === 'function') {
      addItem({ id, name, expenses, incomes })
    }
  }

  return (
    <div className='container'>
      <div className='row'>
        <form
          onSubmit={_addItem}
          className="col-md-12 d-flex align-items-end mt-3 justify-content-center"
          data-testid='form_for_add'>
          <div className="form-group mr-1">
            <label >Item name</label>
            <input
              type="string"
              className="form-control"
              value={name}
              onChange={e => setName(e.target.value)}
            />
          </div>
          <div className="form-group mr-1">
            <label >expenses</label>
            <input
              type="number"
              className="form-control"
              value={expenses}
              onChange={e => setExpenses(e.target.value)}
            />
          </div>
          <div className="form-group mr-1">
            <label >incomes</label>
            <input
              type="number"
              className="form-control"
              value={incomes}
              onChange={e => setIncomes(e.target.value)}
            />
          </div>
          <div className="form-group">
            <button type="submit" className="btn btn-primary">+</button>
          </div>
        </form >
      </div>
    </div>

  )
} 