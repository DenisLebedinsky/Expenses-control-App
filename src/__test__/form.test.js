import React from "react";
import Form from '../components/Form'
import {
  render,
  cleanup,
  waitForElement,
  fireEvent
} from "@testing-library/react";

describe("Form", () => {
  afterEach(cleanup);


  it("test for render data", async () => {

    const { container, getByTestId } = render(<Form />);

    const iputName = await waitForElement(() =>
      getByTestId("form_for_add")
    );

    //console.log(iputName)
    expect(container.firstChild).toMatchSnapshot();
  });

  afterEach(cleanup);

  test('loads items eventually', async () => {
    const { findByText, getByTestId } = render(<Form />)

    fireEvent.change(getByTestId('input_name'), { target: { value: 'a' } })

    expect(findByText('a')).toBeTruthy()
  })
});