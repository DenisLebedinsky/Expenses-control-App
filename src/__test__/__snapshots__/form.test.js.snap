// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test for render data 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <form
      class="col-md-12 d-flex align-items-end mt-3 justify-content-center"
      data-testid="form_for_add"
    >
      <div
        class="form-group mr-1"
      >
        <label>
          Item name
        </label>
        <input
          class="form-control"
          data-testid="input_name"
          type="string"
          value=""
        />
      </div>
      <div
        class="form-group mr-1"
      >
        <label>
          expenses
        </label>
        <input
          class="form-control"
          type="number"
          value="0"
        />
      </div>
      <div
        class="form-group mr-1"
      >
        <label>
          incomes
        </label>
        <input
          class="form-control"
          type="number"
          value="0"
        />
      </div>
      <div
        class="form-group"
      >
        <button
          class="btn btn-primary"
          type="submit"
        >
          +
        </button>
      </div>
    </form>
  </div>
</div>
`;
